### PR TITLE
ZENKO-3106 add release stage

### PR DESCRIPTION
**Pull request by tcarmet**

### CODE REVIEW

File: docs/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Cloudserver Release Plan
+
+## Docker Image generation
```
Comment (wabernat) [inline docs/RELEASE.md]:
```suggestion
## Docker Image Generation
```

File: docs/RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+Docker images are hosted on [registry.scality.com](registry.scality.com).
+It has two names spaces for cloudserver:
```
Comment (wabernat) [inline docs/RELEASE.md]:
```suggestion
CloudServer has two namespaces there:
```

File: docs/RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+* Production Namespace: registry.scality.com/cloudserver
+* Dev Namespace: registry.scality.com/cloudserver-dev
+
+The CI will push images with every CI build tagging the
```
Comment (wabernat) [inline docs/RELEASE.md]:
```suggestion
With every CI build, the CI will push images, tagging the
```

File: docs/RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+content with the developer’s branch short SHA-1 commit hash.
```
Comment (wabernat) [inline docs/RELEASE.md]:
```suggestion
content with the developer branch's short SHA-1 commit hash.
```

File: docs/RELEASE.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+This allows those images to be used by developers, CI builds,
+build chain and so on.
+
+Tagged versions of cloudserver will be stored in the production namespace.
+
+## How to pull docker images
```
Comment (wabernat) [inline docs/RELEASE.md]:
```suggestion
How to Pull Docker Images
```

File: docs/RELEASE.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+```sh
+docker pull registry.scality.com/cloudserver-dev/cloudserver:<commit hash>
+docker pull registry.scality.com/cloudserver/cloudserver:<tag>
+```
+
+## Release Process
+
+To release a production image:
+
+* Chose the name of the tag for the repository and the docker image.
```
Comment (wabernat) [inline docs/RELEASE.md]:
```suggestion
* Name the tag for the repository and Docker image.
```

File: docs/RELEASE.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+* Update the `package.json` using the command `yarn version` with the same tag.
```
Comment (wabernat) [inline docs/RELEASE.md]:
```suggestion
* Use the `yarn version` command with the same tag to update `package.json`.
```

File: docs/RELEASE.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+* Create a PR and merge the `package.json` change.
+
+* Tag the repository using the same tag.
+
+* With the below parameters, [force a build](https://eve.devsca.com/github/scality/cloudserver/#/builders/bootstrap/force/force):
```
Comment (wabernat) [inline docs/RELEASE.md]:
```suggestion
* [Force a build](https://eve.devsca.com/github/scality/cloudserver/#/builders/bootstrap/force/force) using: 
```

File: docs/RELEASE.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  * A given branch that ideally match the tag.
```
Comment (wabernat) [inline docs/RELEASE.md]:
```suggestion
  * A given branch that ideally matches the tag.
```

File: docs/RELEASE.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  * Use the `release` stage.
```
Comment (wabernat) [inline docs/RELEASE.md]:
```suggestion
  * The `release` stage.
```

File: docs/RELEASE.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  * Extra property with name as `tag` and the value as the actual tag.
```
Comment (wabernat) [inline docs/RELEASE.md]:
```suggestion
  * An extra property with the name `tag` and its value being the actual tag.
```

File: eve/main.yml [hunk 12]
```diff
@@ -394,65 +422,66 @@ stages:
       - Upload: *upload-artifacts
       - Upload: *upload-junits
 
-  post-merge:
-    worker:
+  # The docker-build stage will ensure that your images is built on every commit
```
Comment (wabernat) [inline eve/main.yml]:
```suggestion
# The docker-build stage ensures that your images are built on every commit
```

File: eve/main.yml [hunk 13]
```diff
@@ -394,65 +422,66 @@ stages:
...
+  # and also hosted on the registry to help you pull it up and
+  # test it on a real environment if needed.
```
Comment (wabernat) [inline eve/main.yml]:
```suggestion
  # test it in a real environment if needed.
```

File: eve/main.yml [hunk 14]
```diff
@@ -394,65 +422,66 @@ stages:
...
+  # It will also allow us to pull it and rename it when performing a release.
```
Comment (wabernat) [inline eve/main.yml]:
```suggestion
  # It also allows us to pull and rename it when performing a release.
```

File: eve/main.yml [hunk 15]
```diff
@@ -394,65 +422,66 @@ stages:
...
+            --tag=${DEVELOPMENT_DOCKER_IMAGE_NAME}:%(prop:commit_short_revision)s
+          env: *docker_env
+          haltOnFailure: true
+      - ShellCommand:
+          name: push docker image into the development namespace
+          command: docker push ${DEVELOPMENT_DOCKER_IMAGE_NAME}
+          haltOnFailure: true
+          env: *docker_env
+
+  # This stage can be used to release your docker image.
```
Comment (wabernat) [inline eve/main.yml]:
```suggestion
  # This stage can be used to release your Docker image.
```

File: eve/main.yml [hunk 16]
```diff
@@ -394,65 +422,66 @@ stages:
...
+  # To use this stage:
+  # 1. Tag the repository
+  # 2. Force a build with the following parameter:
```
Comment (wabernat) [inline eve/main.yml]:
```suggestion
  # 2. Force a build using:
```

File: eve/main.yml [hunk 17]
```diff
@@ -394,65 +422,66 @@ stages:
...
+  #    * A given branch that ideally match the tag
```
Comment (wabernat) [inline eve/main.yml]:
```suggestion
  #    * A branch that ideally matches the tag
```

File: eve/main.yml [hunk 18]
```diff
@@ -394,65 +422,66 @@ stages:
...
+  #    * Use the release stage
```
Comment (wabernat) [inline eve/main.yml]:
```suggestion
  #    * The release stage
```

File: eve/main.yml [hunk 19]
```diff
@@ -394,65 +422,66 @@ stages:
...
+  #    * Extra property named tag with the value of the name of the tag itself.
```
Comment (wabernat) [inline eve/main.yml]:
```suggestion
  #    * An extra property with the name `tag` and its value being the actual tag.
```

File: docs/RELEASE.md (hunk 5 above)
Comment (tcarmet) [reply docs/RELEASE.md]: Any reason to remove the `##` ?

Comment (tcarmet) [reply docs/RELEASE.md]:
```suggestion
## How to Pull Docker Images
```

**Review by miniscruff — APPROVED** [review]

**Review by alexanderchan-scality — APPROVED** [review]

**Pull request merged**